release/pipeline/configs/environment_config.py

```diff
@@ -15,44 +15,50 @@
 
 EMAIL_LIST = ['istio-release@google.com']
 
-#Make the release on the 3ed Thursday of the mounth.
-#It will run on at 17:15 GST
-MONTHLY_RELEASE_TRIGGER = '15 17 * * 4#3'
-
 AIRFLOW_CONFIG = dict(
-    PROJECT_ID='istio-release',
-    RELEASE_PROJECT_ID='istio-io',
-    MFEST_URL='https://github.com/istio/green-builds',
-    MFEST_FILE='build.xml',
-    MFEST_COMMIT='master@{{{timestamp}}}',
-    VERSION='{major}.{minor}.{patch}-pre{date}-{rc}',
+    BRANCH='{branch}',
+    CHECK_GREEN_SHA_AGE='true',
+    GCR_STAGING_DEST='istio-release',
     GCS_BUILD_BUCKET='istio-release-pipeline-data',
+    GCS_BUILD_PATH='{gcs_build_bucket}/{gcs_path}',
+    GCS_FULL_STAGING_PATH='{gcs_staging_bucket}/{gcs_path}',
+    GCS_RELEASE_TOOLS_PATH='{gcs_build_bucket}/release-tools/{gcs_path}',
     GCS_STAGING_BUCKET='istio-prerelease',
-    GCS_DAILY_PATH='daily-build/{VERSION}',
-    GCS_MONTHLY_STAGE_PATH='prerelease/{VERSION}',
-    GCS_MONTHLY_RELEASE_PATH='istio-release/releases/{VERSION}',
-    SVC_ACCT='202987436673-compute@developer.gserviceaccount.com',
-    TOKEN_FILE='/var/run/secrets/kubernetes.io/serviceaccount/tokenFile',
+    GCS_STAGING_PATH='{gcs_path}',
     GITHUB_ORG='istio',
-    PROD_GITHUB_ORG='istio',
     GITHUB_REPO='istio',
-    GCR_STAGING_DEST='istio-release',
-    GCR_RELEASE_DEST='istio-io',
-    GCS_GITHUB_PATH='istio-secrets/github.txt.enc',
-    DOCKER_HUB='istio')
+    ISTIO_REPO='https://github.com/{github_org}/{github_repo}.git',
+    MFEST_COMMIT='{mfest_commit}',
+    MFEST_FILE='build.xml',
+    MFEST_URL='https://github.com/istio/green-builds',
+    PROJECT_ID='istio-release',
+    SVC_ACCT='202987436673-compute@developer.gserviceaccount.com',
+    TOKEN_FILE='/var/run/secrets/kubernetes.io/serviceaccount/tokenFile',
+    VERIFY_CONSISTENCY='{verify_consistency}',
+    VERSION='{version}')
 
 
-def get_airflow_config(version, timestamp, major, minor, patch, date, rc):
+def get_default_config(branch, gcs_path, mfest_commit,
```

Comment:
This is a rather error prone and non trivial function.

We need a helper function to evaluate the "variables" in the dict generically, instead of resolving selected variables like this.

Also, the caller may as well take AIRFLOW_CONFIG and override only the key-value pairs it cares about, instead of calling this function and provide a few selected values. When we need more, we will need to change this interface.

I think we can either use the totally flexible design (i.e. just a map) or a more structured design (i.e. config builder paradigm).